### PR TITLE
Add initial build insructions to README

### PR DESCRIPTION
**Pull request by marnovandermaas**
It's good to have these build instructions in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,102 @@
 # Overview
 
-This repository contains software, build flows and examples for the Sonata System (https://github.com/lowRISC/sonata-system)
+This repository contains software, build flows and examples for the Sonata System (https://github.com/lowRISC/sonata-system).
+
+## Installing Nix
```
Comment (GregAC) [inline README.md:5]: Shall we put the windows instructions in as well?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,102 @@
...
+
+It'll take a few minutes to create, and you will eventually see a vscode UI. In the bottom panel (which has the terminal) you'll see a + icon in the top right of the panel. Click this and you should see a new terminal open that gives the prompt 'Sonata /workspaces/sonata-software>'. This is the same shell environment you get with the local setup described above.
+
+You can now follow the first build instructions below but you don't need to checkout the repository (the github codespace has done that for you).
+
+Once you have a successful build we recommend either deleting or stopping the codespace to avoid using up any more compute, you can do this from [https://github.com/codespaces](https://github.com/codespaces). Click the three horizontal dots next to the codespace name and choose either 'Delete' or 'Stop Codespace'. A stopped codespace can be started again where if you delete you will need to recreate it for the hackathon.
```
[the remark this answers is not in the document]
Comment (marnovandermaas) [reply README.md]: Thanks for spotting this. This is now reworded.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,102 @@
...
+
+After running this you should see the build run to completion and report success, the critical lines indicating a successful build are:
+
+```sh
+Converted to uf2, output size: 74752, start address: 0x101000
+Wrote 74752 bytes to build/cheriot/cheriot/release/sonata_simple_demo.uf2
+[100%]: build ok, spent 6.827s
+```
+
+(Note output size may differ)
```
Comment (HU90m) [inline README.md:99]:
```suggestion
*Note, the output size may differ.*
```

File: README.md [hunk 4]
```diff
@@ -1,6 +1,102 @@
...
+git clone --recurse-submodule \
+    https://github.com/lowRISC/sonata-software.git
+cd sonata-software
+```
+
+Configure and run build:
+
+```sh
+xmake config --board=sonata
+xmake build
```
Comment (HU90m) [inline README.md:88]:
```suggestion
xmake
```

File: README.md [hunk 5]
```diff
@@ -1,6 +1,102 @@
...
+> sudo systemctl restart nix-daemon
+> ```
+
+## Enter the CHERIoT development environment
+
+Running the following will put you in a shell environment with all the applications needed to build the CHERIoT RTOS.
+
+```sh
+# Enter the shell with
+nix develop github:lowRISC/sonata-software
```
Comment (HU90m) [inline README.md:43]:
```suggestion
nix develop
```

File: README.md [hunk 6]
```diff
@@ -1,6 +1,102 @@
...
+These applications are layered on top of your usual environment. You can see what was added with `echo $PATH`.
+
+```
+do you want to allow configuration setting 'extra-substituters' to be set to 'https://nix-cache.lowrisc.org/public/' (y/N)? y
+do you want to permanently mark this value as trusted (y/N)? y
+do you want to allow configuration setting 'extra-trusted-public-keys' to be set to 'nix-cache.lowrisc.org-public-1:O6JLD0yXzaJDPiQW1meVu32JIDViuaPtGDfjlOopU7o=' (y/N)? y
+do you want to permanently mark this value as trusted (y/N)? y
+warning: ignoring untrusted substituter 'https://nix-cache.lowrisc.org/public/', you are not a trusted user.
+```
+
```
Comment (HU90m) [inline README.md:57]:
````suggestion

For Linux users, there is also an environment with the sonata simulator included:
```
nix develop .#env-with-sim
sonata-simulator --help
```
````

**Review by HU90m — APPROVED** [review]
Happy for this to go in as is. All comments are just suggestions.

File: README.md (hunk 6 above)
Comment (marnovandermaas) [reply README.md:57]: Thanks added.

File: README.md (hunk 5 above)
Comment (marnovandermaas) [reply README.md:43]: I guess this would require them to run this command from within the cloned repository?

File: README.md (hunk 4 above)
Comment (marnovandermaas) [reply README.md]: Thanks changed.

**Pull request merged**